src/Microsoft.AspNetCore.Razor.Design/Internal/AssemblyTagHelperDescriptorResolver.cs

```diff
@@ -65,5 +100,42 @@ protected virtual IEnumerable<Type> GetTagHelperTypes(string assemblyName, Error
         {
             return _tagHelperTypeResolver.Resolve(assemblyName, SourceLocation.Zero, errorSink);
         }
+
+        private void LoadMvcAssembly()
+        {
+            const string typeFullName = "Microsoft.AspNetCore.Mvc.DesignTimeMvcServiceCollectionProvider";
+            const string assemblyName = "Microsoft.AspNetCore.Mvc";
+            const string methodName = "PopulateServiceCollection";
+
+            _loadedAssembly = true;
+
+            try
+            {
+                var providerClass = Type.GetType($"{typeFullName}, {assemblyName}", throwOnError: true);
+
+                _serviceCollectionProviderClassInstance = Activator.CreateInstance(providerClass);
+
+                // Get the method from the type
+                _populateMethod = providerClass.GetMethods(BindingFlags.Public | BindingFlags.Static)
+                    .FirstOrDefault(methodInfo =>
+                    {
+                        if (string.Equals(methodName, methodInfo.Name, StringComparison.Ordinal)
+                            && methodInfo.ReturnParameter.ParameterType.Equals(typeof(void)))
```

Comment:
Probably don't need to check for return type since overloads can only vary by parameters.